integration_test/cases/error_on_connect_test.exs

```diff
@@ -0,0 +1,23 @@
+defmodule ErrorOnConnectTest do
```

Comment:
This test won't run. I think we should rename `backoff_test` to `connect_test` and add this test to that module. Be sure to update `integration_test/tests.exs`.